src/sinks/datadog/logs/service.rs

```diff
@@ -145,6 +156,10 @@ impl Service<LogApiRequest> for LogApiService {
         let mut http_request = http_request.header(CONTENT_LENGTH, request.body.len());
 
         if let Some(headers) = http_request.headers_mut() {
+            for (name, value) in &self.default_headers {
```

Comment:
This is the same as before so I don't think we need to change this as part of this PR, but it did bring to my attention that users could override these headers via `request.headers` and I'm not sure if we want to allow that 🤔 

We could swap the order and apply these after the user-provided headers.

Reply:
I agree, it was originally in this order so that OP could overwrite the value in their config. That is no longer relevant though so I'll swap the order 👍🏻

Reply:
Done.